### PR TITLE
Lava fixes to enable large convolutional networks

### PR DESCRIPTION
**Pull request by bamsumit**
Issue Number: 806, 807

Objective of pull request:

Fixes to enable large convolutional networks
1. Mapper to enable multichip networks
2. Fixes to `io.encoder.DeltaEncoder`
3. `io.extractor.VarWire` to read Lava Var directly from python.

## Pull request checklist

Your PR fulfills the following requirements:
-   [x] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
-   [x] Tests are part of the PR (for bug fixes / features)
-   [x] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
-   [x] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
-   [x] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
-   [x] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
-   [x] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

-   [x] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] Documentation changes
-   [ ] Other (please describe): 

## What is the current behavior?

- 

## What is the new behavior?

-

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: src/lava/proc/conv/utils.py
Comment (timcheck) [inline src/lava/proc/conv/utils.py:1]: Do we want to hardcode TORCH_IS_AVAILABLE = False?

Comment (timcheck) [inline src/lava/proc/conv/utils.py:1]: Do we want to hardcode TORCH_IS_AVAILABLE = False?

File: src/lava/proc/io/encoder.py [hunk 1]
```diff
@@ -188,31 +203,39 @@ def encode_delta_sparse_8(self, s_out):
         if len(idx) == 0:
             idx = np.array([0])
             data = np.array([0])
+        max_idx = 0xFF
```
Comment (timcheck) [inline src/lava/proc/io/encoder.py:206]: Why is max_idx 0xFF?

File: src/lava/proc/io/encoder.py [hunk 2]
```diff
@@ -58,10 +58,13 @@ class DeltaEncoder(AbstractProcess):
...
     vth: int or float
         Threshold of the delta encoder.
-    spike_exp: int
+    spike_exp: Optional[int]
         Scaling exponent with base 2 for the spike message.
         Note: This should only be used for fixed point models.
         Default is 0.
+    num_bits: Optional[int]
+        Precision for spike output. It is applied before spike_exp. If None,
+        precision is not enforced. Default is None.
```
Comment (timcheck) [inline src/lava/proc/io/encoder.py:67]: What does "precision is not enforced" mean? (What will the behavior be?)

Comment (bamsumit) [reply src/lava/proc/io/encoder.py:67]: Means the output will be unbounded.

File: src/lava/proc/io/encoder.py (hunk 1 above)
Comment (bamsumit) [reply src/lava/proc/io/encoder.py:206]: Its delta_8 encoding. Max value is 255.

**Review by timcheck — CHANGES_REQUESTED** [review]
Please address my comments

**Review by timcheck — APPROVED** [review]
Thanks!

**Review by joyeshmishra — APPROVED** [review]

**Pull request merged**